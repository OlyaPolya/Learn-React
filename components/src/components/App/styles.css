* {
  list-style: none;
  margin: 0;
  box-sizing: border-box;
  scroll-behavior: smooth;
}

:root {
  --background: #fbfbfb;
}

ul,
li {
  padding: 0;
}

input,
button,
textarea,
select {
  font: inherit;
}

a {
  text-decoration: none;
  color: inherit;
}

body {
  display: flex;
  justify-content: center;
  background-color: var(--background);
}

#root {
  width: 100%;
}

.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: 'Montserrat', sans-serif;
}

.search-block {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 50%;
}
