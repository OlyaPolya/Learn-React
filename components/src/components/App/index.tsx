import React from 'react';
import './styles.css';
import Search from '../../pages/Search';
import About from '../../pages/About';
import NotFoundPage from '../../pages/NotFound';
import Header from '../Header';
import { Route, Routes } from 'react-router-dom';
import Reviews from '../../pages/Reviews';
import Photo from '../../pages/Photo';
import { Provider } from 'react-redux';
import { setupStore } from 'store/slice';
import { LINKS, Paths } from '../../constants';

const store = setupStore();

const App: React.FC = () => (
  <div className="App">
    <Provider store={store}>
      <Header links={LINKS} />
      <Routes>
        <Route path={Paths.home} element={<Search />} />
        <Route path={Paths.getPhotoPath('*')} element={<Photo />} />
        <Route path={Paths.about} element={<About />} />
        <Route path={Paths.reviews} element={<Reviews />} />
        <Route path={Paths.notFound} element={<NotFoundPage />} />
      </Routes>
    </Provider>
  </div>
);

export default App;
