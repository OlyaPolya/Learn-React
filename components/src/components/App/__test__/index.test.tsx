import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import App from '..';
import { MemoryRouter } from 'react-router-dom';

describe('<App />', () => {
  it('should match snapshot', () => {
    const component = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    expect(component.container).toMatchSnapshot();
  });
});
