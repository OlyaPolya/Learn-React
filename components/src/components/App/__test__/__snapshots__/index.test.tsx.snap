// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="header"
    >
      <p
        class="header-title"
      >
        Learning React Project
      </p>
      <ul
        class="header-links"
      >
        <li
          class="link"
        >
          <a
            href="/"
          >
            Search
          </a>
        </li>
        <li
          class="link"
        >
          <a
            href="/about"
          >
            About Us
          </a>
        </li>
        <li
          class="link"
        >
          <a
            href="/reviews"
          >
            Reviews Form
          </a>
        </li>
      </ul>
    </header>
    <div
      class="search-block"
    >
      <div
        class="search-list"
      >
        <div
          class="search"
        >
          <input
            aria-label="search-input"
            class="search-input"
            placeholder="Nature..."
            type="text"
          />
          <button
            class="search-button"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
